src/internal/miscutil/miscutil.go

```diff
@@ -12,15 +12,13 @@ func WithPipe(wcb func(w io.Writer) error, rcb func(r io.Reader) error) error {
 	pr, pw := io.Pipe()
 	eg := errgroup.Group{}
 	eg.Go(func() error {
-		if err := wcb(pw); err != nil {
-			return pw.CloseWithError(err)
-		}
-		return pw.Close()
+		err := wcb(pw)
+		pw.CloseWithError(err)
+		return err
 	})
 	eg.Go(func() error {
-		if err := rcb(pr); err != nil {
-			return pr.CloseWithError(err)
-		}
+		err := rcb(pr)
+		pr.CloseWithError(err)
 		return pr.Close()
```

Comment:
This seems like it should be `return err`. The pipe reader is already closed at this point and we want the error.